### PR TITLE
Refactor Whitelist data access

### PR DESCRIPTION
**Pull request by eperott**
In preparation for improved operation filter.
Consistent interface with IResource type for whitelist resources in the interface.
consistent interface with RoleResource type for grantee in the interface.

### CODE REVIEW

File: src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/ResourceFactory.java [hunk 1]
```diff
@@ -53,4 +62,9 @@ public static IResource toResource(String resourceName)
                 throw new IllegalArgumentException("Invalid resource type: " + resourceName);
```
Comment (tobier) [inline src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/ResourceFactory.java]: Not part of this change, but what happens if this occur? Will it crash the process and therefor it affect traffic, or will it perhaps crash the plugin and therefore is a vector for disabling whitelisting?

Comment (eperott) [reply src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/ResourceFactory.java]: Ended up creating issue #44 to follow up on this.
